src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Android.cs

```diff
@@ -0,0 +1,473 @@
+
+        public AndroidTzData(params string[] paths)
+        {
+            foreach (var path in paths)
+            {
+                if (LoadData(path))
+                {
+                    tzdataPath = path;
+                    return;
+                }
+            }
+
+            tzdataPath = "/";
+            ids = new[]{ "GMT" };
+        }
+
+        private static string GetApexTimeDataRoot()
+        {
+            string? ret = Environment.GetEnvironmentVariable("ANDROID_TZDATA_ROOT");
+            if (!string.IsNullOrEmpty(ret!)) {
+                return ret!;
+            }
+
+            return "/apex/com.android.tzdata";
+        }
+
+        private static string GetApexRuntimeRoot()
+        {
+            string? ret = Environment.GetEnvironmentVariable("ANDROID_RUNTIME_ROOT");
+            if (!string.IsNullOrEmpty(ret!))
+            {
+                return ret!;
+            }
+
+            return "/apex/com.android.runtime";
+        }
+
+        private bool LoadData(string path)
+        {
+            if (!File.Exists(path))
+                return false;
+
+            try
+            {
+                data = File.OpenRead(path);
+            }
+            catch (IOException)
+            {
+                return false;
+            }
+            catch (UnauthorizedAccessException)
+            {
+                return false;
+            }
+
+            try
+            {
+                ReadHeader();
+                return true;
+            }
+            catch
+            {
+                // log something here instead of the console.
+                //Console.Error.WriteLine ("tzdata file \"{0}\" was present but invalid: {1}", path, e);
+            }
+            return false;
+        }
+
+        private unsafe void ReadHeader()
+        {
+            int size   = Math.Max(Marshal.SizeOf(typeof(AndroidTzDataHeader)), Marshal.SizeOf(typeof(AndroidTzDataEntry)));
+            var buffer = new byte[size];
+            var header = ReadAt<AndroidTzDataHeader>(0, buffer);
+
+            header.indexOffset = NetworkToHostOrder(header.indexOffset);
+            header.dataOffset = NetworkToHostOrder(header.dataOffset);
+
+            sbyte* s = (sbyte*)header.signature;
+            string magic = new string(s, 0, 6, Encoding.ASCII);
+
+            if (magic != "tzdata" || header.signature[11] != 0)
+            {
+                var b = new StringBuilder();
+                b.Append("bad tzdata magic:");
+                for (int i = 0; i < 12; ++i) {
+                    b.Append(' ').Append(((byte)s[i]).ToString("x2"));
+                }
+
+                //TODO: Put strings in resource file
+                throw new InvalidOperationException("bad tzdata magic: " + b.ToString());
+            }
+
+            ReadIndex(header.indexOffset, header.dataOffset, buffer);
+        }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2087:UnrecognizedReflectionPattern",
```

Comment:
Where is the reflection pattern which you are supressing?

Reply:
@steveisok

Reply:
Interesting.  For some reason it was needed before and now it is not.

Reply:
I am hitting this error when removing this line.
```
/Users/mdhwang/runtime_droid/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Android.cs(319,17): Trim analysis error IL2087: System.AndroidTzData.ReadAt<T>(Int64,Byte[]): 'structureType' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicConstructors', 'DynamicallyAccessedMemberTypes.NonPublicConstructors' in call to 'System.Runtime.InteropServices.Marshal.PtrToStructure(IntPtr,Type)'. The generic parameter 'T' of 'System.AndroidTzData.ReadAt<T>(Int64,Byte[])' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to. [/Users/mdhwang/runtime_droid/src/libraries/src.proj]
```